cloudsdk/src/main/java/io/particle/android/sdk/cloud/ParticleDevice.java

```diff
@@ -509,10 +536,17 @@ public ParticleDevice createFromParcel(Parcel in) {
         public TypedFakeFile(byte[] bytes) {
             super("application/octet-stream", bytes);
         }
+        public TypedFakeFile(byte[] bytes, String mimeType) {
+            super(mimeType, bytes);
+        }
 
         @Override
         public String fileName() {
-            return "tinker_firmware.bin";
+            return fileName;
+        }
+
+        public void setFileName(String fileName) {
```

Comment:
Instead of a setter, make the field final and just create a new constructor that lets you set the filename when you build the object.